lib/migration/config/v1/virtual_container_host.go

```diff
@@ -0,0 +1,317 @@
+	InfrastructureAdmin []mail.Address
+}
+
+// RawCertificate is present until we add extraconfig support for [][]byte slices that are present
+// in tls.Certificate
+type RawCertificate struct {
+	Key  []byte `vic:"0.1" scope:"secret"`
+	Cert []byte
+}
+
+// CustomerExperienceImprovementProgram provides configuration for the phone home mechanism
+// This is broken out so that we can have more granular configuration in here in the future
+// and so that it is insulated from changes to Virtual Container Host structure
+type CustomerExperienceImprovementProgram struct {
+	// The server target is as follows, where the uuid is the raw number, no dashes
+	// "https://vcsa.vmware.com/ph-stg/api/hyper/send?_v=1.0&_c=vic.1_0&_i="+vc.uuid
+	// If this is non-nil then it's enabled
+	CEIPGateway url.URL
+}
+
+// Resources is used instead of the ResourceAllocation structs in govmomi as
+// those don't currently hold IO or storage related data.
+type Resources struct {
+	CPU     types.ResourceAllocationInfo
+	Memory  types.ResourceAllocationInfo
+	IO      types.ResourceAllocationInfo
+	Storage types.ResourceAllocationInfo
+}
+
+// SetHostCertificate sets the certificate for authenticting with the appliance itself
+func (t *VirtualContainerHostConfigSpec) SetHostCertificate(key *[]byte) {
+	t.ExecutorConfig.Key = *key
+}
+
+// SetName sets the name of the VCH - this will be used as the hostname for the appliance
+func (t *VirtualContainerHostConfigSpec) SetName(name string) {
+	t.ExecutorConfig.Name = name
+}
+
+// SetDebug configures the debug logging level for the VCH
+func (t *VirtualContainerHostConfigSpec) SetDebug(level int) {
+	t.ExecutorConfig.Diagnostics.DebugLevel = level
+}
+
+// SetMoref sets the moref of the VCH - this allows components to acquire a handle to
+// the appliance VM.
+func (t *VirtualContainerHostConfigSpec) SetMoref(moref *types.ManagedObjectReference) {
+	if moref != nil {
+		t.ExecutorConfig.ID = moref.String()
+	}
+}
+
+// AddNetwork adds a network that will be configured on the appliance VM
+func (t *VirtualContainerHostConfigSpec) AddNetwork(net *executor.NetworkEndpoint) {
+	if net != nil {
+		if t.ExecutorConfig.Networks == nil {
+			t.ExecutorConfig.Networks = make(map[string]*executor.NetworkEndpoint)
+		}
+
+		t.ExecutorConfig.Networks[net.Network.Name] = net
+	}
+}
+
+// AddContainerNetwork adds a network that will be configured on the appliance VM
+func (t *VirtualContainerHostConfigSpec) AddContainerNetwork(net *executor.ContainerNetwork) {
+	if net != nil {
+		if t.ContainerNetworks == nil {
+			t.ContainerNetworks = make(map[string]*executor.ContainerNetwork)
+		}
+
+		t.ContainerNetworks[net.Name] = net
+	}
+}
+
+func (t *VirtualContainerHostConfigSpec) AddComponent(name string, component *executor.SessionConfig) {
+	if component != nil {
+		if t.ExecutorConfig.Sessions == nil {
+			t.ExecutorConfig.Sessions = make(map[string]*executor.SessionConfig)
+		}
+
+		if component.Name == "" {
```

Comment:
this is absolutely not good. We are practically ignoring the potential problems just ignoring a name if it is already set. I better panic here is somebody tries to set a different name instead of current one.